sklearn/tree/tests/test_export.py

```diff
@@ -311,6 +312,106 @@ def test_precision():
                              precision + 1)
 
 
+def test_export_ascii_errors():
+    clf = DecisionTreeClassifier(max_depth=2, random_state=0)
+    clf.fit(X, y)
+
+    assert_raise_message(ValueError,
+                         "max_depth bust be > 0, given 0",
+                         export_ascii, clf, max_depth=0)
+    assert_raise_message(ValueError,
+                         "feature_names must contain 2 elements, got 1",
+                         export_ascii, clf, feature_names=['a'])
+    assert_raise_message(ValueError,
+                         "show_value must be 'all', 'only_leaves' or None",
+                         export_ascii, clf, show_value=["a"])
+    assert_raise_message(ValueError,
+                         "spacing must be > 0, given 0",
+                         export_ascii, clf, spacing=0)
+
+
+def test_export_ascii():
+    clf = DecisionTreeClassifier(max_depth=2, random_state=0)
+    clf.fit(X, y)
+
+    expected_report = dedent("""|---feature_1 <= 0.00
+|   |---* (weights: [3.0, 0.0])
+|---feature_1 >  0.00
+|   |---* (weights: [0.0, 3.0])
+""")
+    assert export_ascii(clf, show_class=False) == expected_report
+
+    expected_report = dedent("""|---feature_1 <= 0.00
+|   | (class: -1)
+|   |---* (weights: [3.0, 0.0])
+|---feature_1 >  0.00
+|   | (class: 1)
+|   |---* (weights: [0.0, 3.0])
+""")
+    assert export_ascii(clf, show_class=True) == expected_report
+
+    expected_report = dedent("""|---feature_1 <= 0.00
+|   | (weights: [3.0, 3.0])
```

Comment:
This seems to be in the wrong place. These are the weights before filtering on `feature_1 <= 0.00`

Reply:
Indeed, I'll see what I can do.